src/wp-includes/option.php

```diff
@@ -643,50 +643,143 @@ function wp_load_alloptions( $force_cache = false ) {
 }
 
 /**
- * Loads and primes caches of certain often requested network options if is_multisite().
+ * Primes specific network options for the current network into the cache with a single database query.
  *
- * @since 3.0.0
- * @since 6.3.0 Also prime caches for network options when persistent object cache is enabled.
+ * Only network options that do not already exist in cache will be loaded.
+ *
+ * If site is not multisite, then call wp_prime_option_caches().
+ *
+ * @since 6.6.0
+ *
+ * @see wp_prime_network_option_caches()
+ *
+ * @param string[] $options An array of option names to be loaded.
+ */
+function wp_prime_site_option_caches( array $options = array() ) {
+	wp_prime_network_option_caches( null, $options );
+}
+
+/**
+ * Primes specific network options into the cache with a single database query.
+ *
+ * Only network options that do not already exist in cache will be loaded.
+ *
+ * If site is not multisite, then call wp_prime_option_caches().
+ *
+ * @since 6.6.0
  *
  * @global wpdb $wpdb WordPress database abstraction object.
  *
- * @param int $network_id Optional. Network ID of network for which to prime network options cache. Defaults to current network.
+ * @param int      $network_id ID of the network. Can be null to default to the current network ID.
+ * @param string[] $options    An array of option names to be loaded.
  */
-function wp_load_core_site_options( $network_id = null ) {
+function wp_prime_network_option_caches( $network_id = null, array $options = array() ) {
 	global $wpdb;
 
-	if ( ! is_multisite() || wp_installing() ) {
+	if ( wp_installing() ) {
+		return;
+	}
+
+	if ( ! is_multisite() ) {
+		wp_prime_option_caches( $options );
 		return;
 	}
 
 	if ( empty( $network_id ) ) {
```

Comment:
Nothing is enforcing that $network_id is an `int` in this function even though that's how it's documented. I'd suggest adding a safety check here, like:

```suggestion
	if ( empty( $network_id ) || ! is_int( $network_id ) ) {
```

Or even consider returning early if an invalid value is passed.

Reply:
Good catch. Here is a fix. https://github.com/WordPress/wordpress-develop/pull/6311/commits/482cc816ea9f970611f8058b2a071cefc3cfe337 

Logic is copied from 

https://github.com/WordPress/wordpress-develop/blob/2a21dd9fef93bcb0405d9f189286402b972faabe/src/wp-includes/option.php#L1856-L1865